### DISCUSSION TITLE
"Mark files as viewed" API

### DISCUSSION THREAD

**Discussion by RMacfarlane** [community/community#24778] [Apps, API and Webhooks]
<p>I’m trying to find the API used for marking files as viewed, as described here: <a href="https://github.blog/2019-07-01-mark-files-as-viewed/" rel="nofollow noopener">https://github.blog/2019-07-01-mark-files-as-viewed/</a>. I want to be able to both fetch data about what files have been marked as viewed, and be able to change the viewed status. I’ve looked at both the v3 and v4 documentation and haven’t found anything to do this. Is this a publicly available API?</p>

**Comment by AndreaGriffiths11** [comment]
<p>Hi <a class="mention" href="/rmacfarlane">@rmacfarlane</a>, </p>
<p>Thank for being here! If I’m not mistaken there isn’t an endpoint to support this request at this time, let me do some more digging and come back to you.</p>

**Comment by circlesmiler** [comment]
<p>Hello Andrea,</p>
<p>I wonder if it possible to provide such API. In fact, the iOS GitHub App is available to mark files as reviewed. So it should be a (private) API available? It would be great to have it public, so that other IDEs like intelliJ can support this feature.</p>
<p>I love the web interface, but for very big pull requests, it’s often better to check the code out and see the differences inside of the IDE.</p>
<p>Thank you und best regards,</p>
<p>André</p>

**Comment by nethgato** [comment]
<p>Hey <a class="mention" href="/circlesmiler">@circlesmiler</a> :wave:</p>
<p>Right now, Andrea’s post is accurate; there is no public endpoint for marking files as viewed. Since the OP, we’ve adjusted how we receive feedback.</p>
<p>I would highly recommend submitting your suggestion via this form:</p>
<aside class="onebox githubfolder">
  <header class="source">
      <img src="https://support-assets.github.com/assets/favicon-3dcd2c977a91b2e3f396b300c5ff0d572295ca632ddb7ba1adcacc1ea68dd5c0.ico" class="site-icon" width="16" height="16">
      <a href="https://support.github.com/contact/feedback" target="_blank" rel="noopener">support.github.com</a>
  </header>
  <article class="onebox-body">
    

<h3><a href="https://support.github.com/contact/feedback" target="_blank" rel="noopener">Share feedback - GitHub Support</a></h3>

<p><a href="https://support.github.com/contact/feedback" target="_blank" rel="noopener">//support.github.com/contact/feedback</a></p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>…so that it gets in front of our PMs for review and consideration.</p>
<p>Thank you! :bow:</p>

**Comment by RMacfarlane** [comment, marked as answer]
<p>I’ve just found that the GraphQL API now actually does have a mutation for this:</p>
<aside class="onebox githubfolder">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/graphql/reference/mutations#markfileasviewed" target="_blank" rel="noopener nofollow ugc">docs.github.com</a>
  </header>
  <article class="onebox-body">
    

<h3><a href="https://docs.github.com/en/graphql/reference/mutations#markfileasviewed" target="_blank" rel="noopener nofollow ugc">Mutations - GitHub Docs</a></h3>

<p><a href="https://docs.github.com/en/graphql/reference/mutations#markfileasviewed" target="_blank" rel="noopener nofollow ugc">//docs.github.com/en/graphql/reference/mutations</a></p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>
<aside class="onebox githubfolder">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/graphql/reference/mutations#unmarkfileasviewed" target="_blank" rel="noopener nofollow ugc">docs.github.com</a>
  </header>
  <article class="onebox-body">
    

<h3><a href="https://docs.github.com/en/graphql/reference/mutations#unmarkfileasviewed" target="_blank" rel="noopener nofollow ugc">Mutations - GitHub Docs</a></h3>

<p><a href="https://docs.github.com/en/graphql/reference/mutations#unmarkfileasviewed" target="_blank" rel="noopener nofollow ugc">//docs.github.com/en/graphql/reference/mutations</a></p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>and you can query for this object to find out the viewed status for a file:</p><aside class="onebox githubfolder">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/graphql/reference/objects#pullrequestchangedfile" target="_blank" rel="noopener nofollow ugc">docs.github.com</a>
  </header>
  <article class="onebox-body">
    

<h3><a href="https://docs.github.com/en/graphql/reference/objects#pullrequestchangedfile" target="_blank" rel="noopener nofollow ugc">Objects - GitHub Docs</a></h3>

<p><a href="https://docs.github.com/en/graphql/reference/objects#pullrequestchangedfile" target="_blank" rel="noopener nofollow ugc">//docs.github.com/en/graphql/reference/objects</a></p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>There’s still nothing in the REST API, but this unblocks me!</p>

**Comment by nethgato** [comment]
<p>Thanks for updating this post <a class="mention" href="/rmacfarlane">@rmacfarlane</a> :bow:</p>
<p>I’m so happy to learn alongside you that the <code>markFileAsViewed</code> mutation exists; TIL!</p>
